app/scripts/metamask-controller.js

```diff
@@ -267,6 +268,8 @@ export default class MetamaskController extends EventEmitter {
     // instance of a class that wraps the extension's storage local API.
     this.localStoreApiWrapper = opts.localStore;
 
+    this.currentMigrationVersion = opts.localStore?.meta?.version;
```

Comment:
`localStore` is the local store class, not the state object. We'd need to do something like this instead:

```suggestion
    this.currentMigrationVersion = opts.localStore?.get()?.meta?.version;
```

Reply:
Updated to use `this.localStoreApiWrapper` instead, and to remove an unnecessary optional chain operator:
```suggestion
    this.currentMigrationVersion = this.localStoreApiWrapper.get()?.meta?.version;
```

Reply:
Though, maybe better to avoid the additional call to local storage, instead keeping the full persisted state around from when we load it during initialization.  See the `loadStateFromPersistence` function, it returns just the data portion of the persisted state. It could return the full object instead, passing the migration version along into the controller constructor.

Reply:
Took the approach of the most recent comment in 1b88d8c7ed